file/file_instance.go

```diff
@@ -72,11 +78,87 @@ func (f *fileInstance) GetDataRecords() []lib.Record {
 
 // GeneratorTrailer returns trailer segment that created automatically
 func (f *fileInstance) GeneratorTrailer() (lib.Record, error) {
-	return nil, nil
+	var trailer lib.Record
+	var information *lib.TrailerInformation
+	var err error
+	if f.format == PackedFileFormat {
+		trailer = lib.NewPackedTrailerRecord()
+		information, err = f.generatorPackedTrailer()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		trailer = lib.NewTrailerRecord()
+		information, err = f.generatorTrailer()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	fromFields := reflect.ValueOf(information).Elem()
+	toFields := reflect.ValueOf(trailer).Elem()
+	for i := 0; i < fromFields.NumField(); i++ {
+		fieldName := fromFields.Type().Field(i).Name
```

Comment:
Are there `nil` checks to perform here so we avoid a panic?

Reply:
fromFields.Type().Field(i) will return struct, not pointer